x/valset-pref/msg_server_test.go

```diff
@@ -307,7 +307,241 @@ func (s *KeeperTestSuite) TestDelegateToValidatorSet() {
+// 			// setup message server
+// 			msgServer := valPref.NewMsgServerImpl(s.App.ValidatorSetPreferenceKeeper)
+// 			c := sdk.WrapSDKContext(s.Ctx)
+
+// 			if test.setValSet {
+// 				// SetValidatorSetPreference sets a new list of val-set
+// 				_, err := msgServer.SetValidatorSetPreference(c, types.NewMsgSetValidatorSetPreference(test.delegator, preferences))
+// 				s.Require().NoError(err)
+
+// 				// DelegateToValidatorSet delegate to existing val-set
+// 				_, err = msgServer.DelegateToValidatorSet(c, types.NewMsgDelegateToValidatorSet(test.delegator, test.coinToStake))
+// 				s.Require().NoError(err)
+// 			}
+
+// 			if test.setExistingDelegations {
+// 				err := s.PrepareExistingDelegations(s.Ctx, valAddrs, test.delegator, test.coinToStake.Amount)
+// 				s.Require().NoError(err)
+// 			}
+
+// 			if test.addToNormalStake {
+// 				validator, found := s.App.StakingKeeper.GetValidator(s.Ctx, extraValidator)
+// 				s.Require().True(found)
+
+// 				// Delegate more token to the validator, this means there is existing Valset delegation as well as regular staking delegation
+// 				_, err := s.App.StakingKeeper.Delegate(s.Ctx, test.delegator, test.addToStakeCoins.Amount, stakingtypes.Unbonded, validator, true)
+// 				s.Require().NoError(err)
+// 			}
+
+// 			if test.addToValSetStake {
+// 				valAddr, err := sdk.ValAddressFromBech32(preferences[0].ValOperAddress)
+// 				s.Require().NoError(err)
+
+// 				validator, found := s.App.StakingKeeper.GetValidator(s.Ctx, valAddr)
+// 				s.Require().True(found)
+
+// 				// Delegate more token to the validator, this means there is existing Valset delegation as well as regular staking delegation
+// 				_, err = s.App.StakingKeeper.Delegate(s.Ctx, test.delegator, test.addToStakeCoins.Amount, stakingtypes.Unbonded, validator, true)
+// 				s.Require().NoError(err)
+// 			}
+
+// 			_, err := msgServer.UndelegateFromValidatorSet(c, types.NewMsgUndelegateFromValidatorSet(test.delegator, test.coinToUnStake))
+// 			if test.expectPass {
+// 				s.Require().NoError(err)
+
+// 				// extra validator + valSets
+// 				var vals []sdk.ValAddress
+// 				if test.addToNormalStake {
+// 					vals = []sdk.ValAddress{extraValidator}
+// 				}
+// 				for _, val := range preferences {
+// 					vals = append(vals, sdk.ValAddress(val.ValOperAddress))
+// 				}
+
+// 				var unbondingDelsAmt []sdk.Dec
+// 				unbondingDels := s.App.StakingKeeper.GetAllUnbondingDelegations(s.Ctx, test.delegator)
+// 				for i := range unbondingDels {
+// 					unbondingDelsAmt = append(unbondingDelsAmt, sdk.NewDec(unbondingDels[i].Entries[0].Balance.Int64()))
+// 				}
+
+// 				sort.Slice(unbondingDelsAmt, func(i, j int) bool {
+// 					return unbondingDelsAmt[i].GT(unbondingDelsAmt[j])
+// 				})
+
+// 				s.Require().Equal(test.expectedSharesToUndelegate, unbondingDelsAmt)
+// 			} else {
+// 				s.Require().Error(err)
+// 			}
+// 		})
+// 	}
+// }
+
+func (s *KeeperTestSuite) TestUnDelegateFromRebalancedValidatorSet() {
```

Comment:
TestUnDelegateFromRebalancedValidatorSet is identical to the commented out UnDelegateFromValidatorSet, because the behavior of UnDelegateFromValidatorSet is technically UnDelegateFromRebalancedValidatorSet.